svc/pkg/mm/util/src/verification.rs

```diff
@@ -149,8 +149,22 @@ pub async fn verify_config(
 					create_config.enable_public,
 					create_config.enable_private,
 				) {
-					(backend::matchmaker::lobby::Publicity::Public, true, _) => {}
-					(backend::matchmaker::lobby::Publicity::Private, _, true) => {}
+					(backend::matchmaker::lobby::Publicity::Public, allowed, _) => {
+						if allowed {
+							bail_with!(
+								MATCHMAKER_CUSTOM_LOBBY_CONFIG_INVALID,
+								reason = r#""public" publicity not allowed with this custom game mode"#
+							);
+						}
+					}
```

Comment:
I could be backwards on this logic, but this looks like it says "if enable public is true and we're on the public publicity check, then throw an error". Should it be `!allowed`? Other than this, PR looks good 👍

Reply:
Good catch lol, crazy how that slipped by. Should also be using `ensure_with`